server/src/test/java/org/elasticsearch/action/admin/indices/create/TransportCreatePartitionsActionTest.java

```diff
@@ -60,116 +55,98 @@ public void prepare() {
-            }
-        });
-
-        assertThat("Create indices action did not lead to a new cluster state",
-            response.get(5, TimeUnit.SECONDS).isAcknowledged(), is(true));
-
-        ClusterState currentState = cluster().clusterService().state();
-        ImmutableOpenIntMap<IndexShardRoutingTable> newRouting = currentState.routingTable().indicesRouting().get("index_0").getShards();
-        assertTrue("[index_0][0] must be started already", newRouting.get(0).primaryShard().started());
-    }
+        execute("create table table1 (a int, b int) " +
+            "partitioned by (a) " +
+            "clustered into 1 shards " +
+            "with (number_of_replicas=0)");
 
-    @Test
-    public void testCreateBulkIndicesIgnoreExistingSame() throws Exception {
-        PutIndexTemplateRequest templateRequest = new PutIndexTemplateRequest("*")
-            .patterns(List.of("*"))
-            .mapping(Map.of())
-            .settings(Settings.builder()
-                .put("number_of_shards", 1)
-                .put("number_of_replicas", 0)
-            );
-        cluster().client().execute(PutIndexTemplateAction.INSTANCE, templateRequest).get();
-        List<String> indices = Arrays.asList("index1", "index2", "index3", "index1");
-        AcknowledgedResponse response = action.execute(
-            new CreatePartitionsRequest(indices)
-        ).get();
-        assertThat(response.isAcknowledged(), is(true));
+        ensureYellow();
         Metadata indexMetadata = cluster().clusterService().state().metadata();
-        for (String index : indices) {
-            assertThat(indexMetadata.hasIndex(index), is(true));
+
+        // CREATE TABLE... PARTITIONED BY doesn't create an index, it creates only template via MetadataIndexTemplateService.
+        assertThat(indexMetadata.indices()).isEmpty();
+
+        // Create some indices/partitions
+        execute("insert into table1 (a,b) values (1,1), (2,2), (3,3)");
+        execute("refresh table table1");
+        Metadata updatedMetadata = cluster().clusterService().state().metadata();
+        assertThat(updatedMetadata.indices().size()).isEqualTo(3); // 1 table with 3 partitions.
+
+        // Try to insert into same partitions, when index is created it should not be re-created
+        // Only 1 new index should be created here, for a partition 4.
+        execute("insert into table1 (a,b) values (1,1), (2,2), (3,3), (4,4)");
+        execute("refresh table table1");
+
+        ClusterState currentState = cluster().clusterService().state();
+        updatedMetadata = currentState.metadata();
+        assertThat(updatedMetadata.indices().size()).isEqualTo(4); // 1 table with 4 partitions.
+        for (ObjectCursor<IndexRoutingTable> cursor: currentState.routingTable().indicesRouting().values()) {
+            IndexRoutingTable routingTable = cursor.value;
+            assertThat(routingTable.allPrimaryShardsActive()).isTrue();
```

Comment:
This used to be
```
ImmutableOpenIntMap<IndexShardRoutingTable> newRouting = currentState.routingTable().indicesRouting().get("index_0").getShards();
assertTrue("[index_0][0] must be started already", newRouting.get(0).primaryShard().started());
```
I think indices size assertion is enough on its own: initially was 3 indices, after inserting into 4 partitions we don't have 7 (would have been the case if we overwrote) but just 4.


Do I get it right, that assertion tries to track [bug](https://github.com/crate/crate/commit/d3c6d8dd68404ae6620eed0daf32bdc1c4066e35)
insert into existing partition -> re-recreate index -> index is now empty/data is lost -> shard is inactive.

I.e is index being just created/empty makes its primary shard inactive?

Reply:
Thanks for the pointer! Updated test in https://github.com/crate/crate/pull/13807/commits/154d08318d2c85e55ee19c4c6a2f2c3a458b77ea